salt/_pillar/metalk8s_private.py

```diff
@@ -35,14 +36,14 @@ def _read_apiserver_key():
     #   2- Salt master Pod runs on the 'bootstrap' minion, which is, by chance,
     #      also the initial master node
 
-    if not os.path.isfile(APISERVER_KEY_PATH):
+    try:
```

Comment:
Actually, the real equivalent would be:

```python
    try:
        with open(APISERVER_KEY_PATH, "r") as config_file:
            encryption_key = config_file.read()
    except FileNotFoundError:
        return {"_errors": [
            "Missing secrets encryption key to share with master node."
        ]}
```

Just catch file missing with a proper errors, and let other exceptions bubble up.

Reply:
Meh, no `FileNotFoundError` in Python 2, so it would be more like:

```python
    try:
        with open(APISERVER_KEY_PATH, "r") as config_file:
            encryption_key = config_file.read()
    except IOError as exn:
        if exn.errno == errno.ENOENT:
            return {"_errors": [
                "Missing secrets encryption key to share with master node."
            ]}
```